### DISCUSSION TITLE
Add custom url routes?

### DISCUSSION THREAD

**Discussion by ctwhome** [TerriaJS/terriajs#5445] [Q&A]
Hi! 

Is there a way to add 'react-router-dom' to terria to allow 'historyApiFallback' so we can use custom routes on the URL to switch between react components? or there is a different way terria is already using? 

Something like this https:// domain/builder.

Is it possible to add the configuration to the terriajs server?

Using # in the URL triggers this call 'http://localhost:3000/builder' triggers 
a Request URL: http://localhost:3000/init//builder.json in the network tab and an error of 4040 not found in the console

![image](https://user-images.githubusercontent.com/4195550/115842264-ef41b600-a41d-11eb-969b-6f80be32e241.png)

**Comment by zoran995** [comment]
TerriaJS team is working on routing for some time, you can check #4961 for more details

**Comment by ctwhome** [comment]
thanks for the answers, but I am afraid that is for the 'next' version, we are using the current 7.x.x
